sdk/iot/Azure.Iot.Hub.Service/src/Authentication/SasTokenProviderWithSharedAccessKey.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Azure.Iot.Hub.Service.Authentication
+{
+    public class SasTokenProviderWithSharedAccessKey : ISasTokenProvider
+    {
+        private static readonly TimeSpan s_defaultTimeToLive = TimeSpan.FromMinutes(30);
+        private readonly object _lock = new object();
+
+        private readonly string _hostname;
+        private readonly string _sharedAccessPolicy;
+        private readonly string _sharedAccessKey;
+        private readonly TimeSpan _timeToLive;
+
+        private string _cachedSasToken;
+        private DateTimeOffset _tokenExpiryTime;
+
+        // Private constructor to prevent accidental initialzation.
+        private SasTokenProviderWithSharedAccessKey()
```

Comment:
There is already a parameterized constructor so why do you need this?

Reply:
`// Private constructor to prevent accidental initialzation.`

Reply:
Aren't these normally protected, to allow for mocking?

Reply:
To clarify, if there is a parameterized constructor, how can we accidentally initialize? The other constructor is still public. Maybe i'm missing something here. But yes, to Tim's point we might need a protected one for mocking.

Reply:
you are absolutely right @vinagesh , I don't know what I was thinking!
I'll create a protected constructor, for mocking.